src/stactools/sentinel2/stac.py

```diff
@@ -47,7 +47,16 @@ def create_item(
     granule_metadata = GranuleMetadata(safe_manifest.granule_metadata_href,
                                        read_href_modifier)
 
-    item = pystac.Item(id=product_metadata.product_id,
+    # Remove the processing number and SAFE Product Discriminator
+    # See https://sentinel.esa.int/web/sentinel/user-guides/sentinel-2-msi/naming-convention
+    id_parts = product_metadata.product_id.split('_')
+    # Remove <Product Discriminator>.SAFE
+    id_parts = id_parts[:-1]
```

Comment:
Are there any code paths where the last element could not be `.SAFE`? If there could be, what do you think about an `assert` here to blow people up early rather than letting something sneak through?

Reply:
The product ID is being pulled from the metadata files; I don't believe there'd be any non-standard product URI values inside that XML file. This follows the [naming conventions](https://sentinel.esa.int/web/sentinel/user-guides/sentinel-2-msi/naming-convention) specified by the ESA. The potentially most option is to write a regex that asserts assumptions about the ID, but honestly I'd be more worried about the regex being wrong than the metadata value. I'll add an assert about the .SAFE ending just to be sure.